### PR TITLE
Feat/slepukhin/user email change

### PR DESCRIPTION
**Pull request by vitslepukhin**

### CODE REVIEW

File: src/app/profile/components/change-email/change-email.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  let fixture: ComponentFixture<ChangeEmailComponent>;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        declarations: [ChangeEmailComponent],
+        imports: [...RootModules(), ComponentTestingModule],
+        providers: [
+          Actions,
+          SelectableCountryOnSearchService,
```
Comment (enkryptor) [inline src/app/profile/components/change-email/change-email.component.spec.ts:23]: what are these dependencies for?

File: src/app/profile/components/change-email/change-email.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      this.resetLoading();
+      this.cd.markForCheck();
+    });
+
+    this.actions$
+      .pipe(ofActionSuccessful(CurrentUserActions.VerifyEmailAction), first())
+      .subscribe((action: CurrentUserActions.VerifyEmailAction) => {
+        this.newEmail = action.verifyEmail.email;
+        this.success = true;
+        this.resetLoading();
```
Comment (enkryptor) [inline src/app/profile/components/change-email/change-email.component.ts:40]: what if the action fails?

File: src/app/profile/components/change-email/change-email.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        this.cd.markForCheck();
+      });
+
+    this.route.queryParams.pipe(takeUntil(this.destroy$)).subscribe((params: VerifyEmail) => {
+      this.setLoading();
```
Comment (enkryptor) [inline src/app/profile/components/change-email/change-email.component.ts:45]: why setLoading() if queryParams were empty?

File: src/app/profile/components/change-email/change-email.component.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly actions$: Actions,
+    private readonly destroy$: NgDestroyService,
+    private readonly cd: ChangeDetectorRef,
+  ) {}
+
+  public ngOnInit(): void {
+    this.actions$.pipe(ofActionErrored(CurrentUserActions.VerifyEmailAction), first()).subscribe(() => {
```
Comment (enkryptor) [inline src/app/profile/components/change-email/change-email.component.ts:29]: I suggest using store.dispatch() switched from queryParams

File: src/app/profile/components/change-email/change-email.component.ts (hunk 3 above)
Comment (vitslepukhin) [reply src/app/profile/components/change-email/change-email.component.ts:45]: If queryParams empty - the action VerifyEmailAction will failure, when resetLoading

**Review by enkryptor — CHANGES_REQUESTED** [review]
let's move from @Dispatch() to using explicit store.dispatch() method

**Review by enkryptor — APPROVED** [review]

**Pull request merged**